test/common/TestBedChild.cpp

```diff
@@ -315,6 +314,14 @@ namespace RcclUnitTesting
 
   ErrCode TestBedChild::ExecuteCollectives()
   {
+    int rankListSize, tempRank;
+    std::vector<int> rankList = {};
+    PIPE_READ(rankListSize);
+
+    for (int rank = 0; rank < rankListSize; ++rank){
+      PIPE_READ(tempRank);
+      rankList.push_back(tempRank);
+    }
```

Comment:
The ranks being passed in are global ranks, not local ranks.
As a result, you should convert them to localRanks (if they are on this child, based on rank offset, number of ranks on this child).
Need to use a more descriptive name rather than rankList to signify what it is doing.  ranksToExecute?  Also, explain what it means if ranksToExecute is empty with a comment.